src/index.ts

```diff
@@ -0,0 +1,77 @@
+const validVersions = [
+	'es3',
+	'es5',
+	'es2015',
+	'es2016',
+	'es2017',
+	'es2018',
+	'es2019',
+	'es2020',
+	'esnext',
+] as const;
+
+export type ValidVersion = typeof validVersions[number];
+
+let desiredVersionIndex = 0;
+
+export const set = (version: ValidVersion) => {
+	const versionIndex = validVersions.indexOf(version);
+	if (versionIndex === -1) {
+		throw new Error('Invalid es version to set: ' + version);
+	}
+	desiredVersionIndex = versionIndex;
+};
+
+export const get = (supportedVersions?: ValidVersion[]) => {
+	if (supportedVersions == null) {
+		// If no supported version array is passed just return the desired version
+		return validVersions[desiredVersionIndex];
+	}
+
+	// Otherwise check for the closest matching supported version and return that, priority is:
+	// 1st: exact match
+	// 2nd: closest match under
+	// 3rd: closest match over
+	// 4th: error
+	let closestVersionUnder: number | undefined;
+	let closestVersionOver: number | undefined;
+	for (const supportedVersion of supportedVersions) {
+		const supportedVersionIndex = validVersions.indexOf(supportedVersion);
```

Comment:
Super-tiny-optional-NIT: That's `O(supportedVersion.length * validVersions.length)` since `Array.indexOf` is `O(n)`.
How about an index/order dictionary (`Dictionary<number>`)?
I don't expect these lists become that large... but this cost can pile up as more libraries w/ balena-es-version are required.